pkg/openstack/loadbalancer.go

```diff
@@ -1885,6 +1882,9 @@ func (lbaas *LbaasV2) ensureOctaviaLoadBalancer(ctx context.Context, clusterName
 			if err := lbaas.ensureOctaviaHealthMonitor(loadbalancer.ID, cutString(fmt.Sprintf("monitor_%d_%s", portIndex, lbName)), pool, port, svcConf); err != nil {
 				return nil, err
 			}
+
+			// After all ports have been processed, remaining listeners are removed if they were created by this Service.
+			curListeners = popListener(curListeners, listener.ID)
```

Comment:
I don't quite understand, why putting this line to the end could solve the issue?

Reply:
because the listener pointer is being used in in the next function but the pointer is poped before so the reference is not longer the right listener.
if you debug the code you will see how the reference is shifted to the wrong listener while the next function needs the right one

Reply:
when you say `next function` are you saying `deleteOctaviaListeners` which is next function?

if so I am also curious why move to here can help ..thanks

Reply:
No, I mean ```lbaas.ensureOctaviaPool(loadbalancer.ID, cutString(fmt.Sprintf("pool_%d_%s", portIndex, lbName)), listener, service, port, nodes, svcConf)``` 
The listener pointer is being used there but the reference is not longer correct!

Reply:
I will look into it in the next days and try to give you feedback

Reply:
it turns out the issue was on the OpenStack API because the backend was in trouble and records where mixed up.
Sorry for the noise, I will close the issue.

Reply:
Hi @FlorinPeter, seems the changes hare is more reasonable for now, but could you please add a comment that describes the relations between curListenerMapping and curListeners, e.g. modifying the curListeners would also change the mapping.